hack/kind-down.sh

```diff
@@ -1,18 +1,29 @@
 #!/usr/bin/env bash
 
-# Copyright 2020 the Pinniped contributors. All Rights Reserved.
+# Copyright 2020-2023 the Pinniped contributors. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
 set -euo pipefail
 
-ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 cd "${ROOT}"
 
 if [[ "${PINNIPED_USE_LOCAL_KIND_REGISTRY:-}" != "" ]]; then
   reg_name='kind-registry.local'
-  docker network disconnect "kind" "${reg_name}" || true
-  docker stop "${reg_name}" || true
-  docker rm "${reg_name}" || true
+
+  # If the container is running...
+  if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)" == 'true' ]; then
+    # Disconnect it from the kind network, if it was connected.
+    if [ "$(docker inspect -f='{{json .NetworkSettings.Networks.kind}}' "${reg_name}")" != 'null' ]; then
+      docker network disconnect "kind" "${reg_name}" >/dev/null
+    fi
+
+    echo "Stopping container $reg_name ..."
```

Comment:
I'm inclined to copy `function log_note()` from `prepare-for-integration-tests` so the output matches color.